File: algorithms/Kadanes Algorithm/Kadanes.py

# Main Function to find Maximum Subarray Sum

def maxSubArraySum(a):
    
    max_now = 0
    total_max = -9999999999  # Initializing variables with 0 and  Negative Infinity

   # Traversing in List/Array

    for i in range(0,len(a)):
        # Finding out Current Maximum Sum in List/Array
        total_max = max(a[i],total_max+a[i])  
        # Comparing the Current Maximum Sum with Previous One
        max_now = max(max_now,total_max)

    # This will return Maximum Subarray Sum
    return max_now 


# Driver Code
a = [-2, -3, 4, -1, -2, 1, 5, -3] 
print("Maxium Subarray sum is",maxSubArraySum(a))

File: algorithms/Kadanes Algorithm/Kadanes.cpp
#include<iostream>
#include<climits>
using namespace std;
// Main Function to find Maximum Subarray Sum
int maxSubarraySum(int a[], int size) 
{ 
   int total_max = INT_MIN;   // Initializing variables with 0 and  Negative Infinity
   int max_now = 0; 
   // Traversing in Array
   for (int i = 0; i < size; i++) 
   {   // Finding out Current Maximum Sum in Array
        max_now = max(a[i], max_now+a[i]); 
         // Comparing the Current Maximum Sum with Previous One
        total_max = max(total_max, max_now); 
   } 
    // This will return Maximum Subarray Sum
   return total_max; 
} 
// Driver Code
int main()
{
      int a[] = {-2, -3, 4, -1, -2, 1, 5, -3}; 
      int size = sizeof(a)/sizeof(a[0]);
      cout<<"Max Subarray Sum is "<<maxSubarraySum(a,size);
}
